### PR TITLE
Helm chart for consent validator

### PR DESCRIPTION
**Pull request by pranavv0**
1. Helm chart for Consent Validator

### CODE REVIEW

File: K8s-deployment/Charts/consent-validator/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.26
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/Chart.yaml:18]:
```suggestion
version: 0.1.0
```

File: K8s-deployment/Charts/consent-validator/.helmignore
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/.helmignore:1]: Can we make this also common for all the Charts?

File: K8s-deployment/Charts/consent-validator/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.0.0"
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/Chart.yaml:23]:
```suggestion
appVersion: "1.0.0-alpha"
```

File: K8s-deployment/Charts/consent-validator/README.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+Helm Chart for IUDX Consent Validator Server Deployment
+
+
+## Generating secrets
+
+Make a copy of sample secrets directory:
+
+```console
+cp -r example-secrets/secrets .
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/README.md:11]: There is no secrets directory inside the example-secrets dir currently, add it.

File: K8s-deployment/Charts/consent-validator/README.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+```
+```
+# secrets directory after generation of secret files
+secrets/
+├── .cv.env
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/README.md:16]: no sample `.cv.env` file present. add it.

File: K8s-deployment/Charts/consent-validator/README.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+└── config.json
+└── keystore
+```
+Configure the env file .config.env in secret/ appropriately
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/README.md:20]: Fix and mention the correct files that need to be configured.

File: K8s-deployment/Charts/consent-validator/README.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+## Define Appropriate values of resources
+
+Define Appropriate values of resources -
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/README.md:23]: Add that the hostname also has to be configured for ingress.

File: K8s-deployment/Charts/consent-validator/example-aws-resource-values.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  hostname: cv-hostname
+  annotations: {
+    cert-manager.io/cluster-issuer: "letsencrypt-staging" # use letsencrypt-prod if proper letsencrypt certs are required
+  }
+
+consentValidator:
+  nodeSelector:
+    node.kubernetes.io/instance-type: t3a.large
+  resources:
+    limits:
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-aws-resource-values.yaml:11]: Too high, set appropriately. Similar to Azure.

File: K8s-deployment/Charts/consent-validator/example-azure-resource-values.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    cert-manager.io/cluster-issuer: "letsencrypt-staging" # use letsencrypt-prod if proper letsencrypt certs are required
+  }
+
+consentValidator:
+  nodeSelector:
+    node.kubernetes.io/instance-type: Standard_D2as_v4
+  resources:
+    limits:
+      cpu: 500m
+      memory: 0.6Gi
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-azure-resource-values.yaml:13]:
```suggestion
      memory: 600Mi
```
Avoid writing memory in fractions, might be problematic in some places.

File: K8s-deployment/Charts/consent-validator/example-azure-resource-values.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    requests: 
+      cpu: 300m
+      memory: 0.4Gi
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-azure-resource-values.yaml:16]:
```suggestion
      memory: 400Mi
```

File: K8s-deployment/Charts/consent-validator/example-secrets/config.json [hunk 10]
```diff
@@ -0,0 +1,44 @@
+{
+    "version": "1.0",
+    "postgresOptions": {
+        "databaseIP": "database.ip",
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-secrets/config.json:4]: You can put the Postgres service here. And for the rest of the config as well, try to set the values which we know won't change. Like the db name, db users.

File: K8s-deployment/Charts/consent-validator/example-secrets/config.json [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        "databasePort": "5433",
+        "databaseName": "db-name",
+        "databaseSchema": "db-schema",
+        "databaseUserName": "db-user",
+        "databasePassword": "db-password",
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-secrets/config.json:9]: add shell command to retrieve the password. Refer to sample configs of other API servers.

File: K8s-deployment/Charts/consent-validator/example-secrets/config.json [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+            "dpNumberKey": "ppbNumber",
+            "dpNameKey": "farmerName",
+            "allDpsNotFoundMsg": "No Data Found"
+        }
+    },
+    "pdfRateLimitOptions": {
+        "rateLimitCount" : 5,
+        "rateLimitDuration" : "P1D"
+    },
+    "catalogueServer": "https://cat.iudx.io",
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-secrets/config.json:35]: Put placeholders for domains. Refer to configs of other API servers.

File: K8s-deployment/Charts/consent-validator/values.yaml [hunk 13]
```diff
@@ -0,0 +1,584 @@
...
+  ## @param consentValidator.livenessProbe.enabled Enable livenessProbe on consentValidator containers
+  ## @param consentValidator.livenessProbe.initialDelaySeconds Initial delay seconds for livenessProbe
+  ## @param consentValidator.livenessProbe.periodSeconds Period seconds for livenessProbe
+  ## @param consentValidator.livenessProbe.timeoutSeconds Timeout seconds for livenessProbe
+  ## @param consentValidator.livenessProbe.failureThreshold Failure threshold for livenessProbe
+  ## @param consentValidator.livenessProbe.successThreshold Success threshold for livenessProbe
+  ## @param consentValidator.livenessProbe.path Path for httpGet
+  ##
+  livenessProbe:
+    enabled: false
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/values.yaml:128]: Why is it disabled?

**Review by karun-singh — CHANGES_REQUESTED** [review]

File: K8s-deployment/Charts/consent-validator/values.yaml (hunk 13 above)
Comment (pranavv0) [reply K8s-deployment/Charts/consent-validator/values.yaml:128]: currently it does not have metrics

File: K8s-deployment/Charts/consent-validator/README.md [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+cp -r example-secrets/secrets .
+```
+```
+# secrets directory after generation of secret files
+secrets/
+├── .cv.env
+└── config.json
+└── keystore
+```
+Configure the env file .cv.env in secret/ appropriately
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/README.md:20]:
```suggestion
Configure the `.cv.env` and `config.json` files in secrets/ appropriately
```

File: K8s-deployment/Charts/consent-validator/example-secrets/config.json (hunk 10 above)
Comment (karun-singh) [reply K8s-deployment/Charts/consent-validator/example-secrets/config.json]: @pranavv0 Please incorporate all config changes as well.

File: K8s-deployment/Charts/consent-validator/values.yaml (hunk 13 above)
Comment (pranavv0) [reply K8s-deployment/Charts/consent-validator/values.yaml:128]: metrics will be added, we can wait till then

File: K8s-deployment/Charts/consent-validator/example-secrets/secrets/config.json [hunk 15]
```diff
@@ -0,0 +1,46 @@
+{
+    "version": "1.0",
+    "postgresOptions": {
+      "databaseIP": "database.iudx.io",
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-secrets/secrets/config.json:4]:
```suggestion
      "databaseIP": "psql-postgresql-ha-pgpool.postgres.svc.cluster.local",
```
Use postgres' service

File: K8s-deployment/Charts/consent-validator/example-secrets/secrets/config.json [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+      "databasePort": "5432",
+      "databaseName": "adex_consent_manager",
+      "databaseSchema": "adexconsent",
+      "databaseUserName": "adex_consent_user",
+      "databasePassword": "<Postgres Password of adex_consent_user >",
```
Comment (karun-singh) [inline K8s-deployment/Charts/consent-validator/example-secrets/secrets/config.json:9]: Replace with the shell command to retrieve the password. 
```suggestion
      "databasePassword": "<psql-pass:- cat iudx-deployment/K8s-deployment/Charts/postgresql/secrets/passwords/postgres-consent-password>",
```

**Review by karun-singh — CHANGES_REQUESTED** [review]

**Review by karun-singh — APPROVED** [review]

**Review by abhi4578 — APPROVED** [review]

**Pull request merged**